caikit_nlp/modules/embedding_retrieval/embedding.py

```diff
@@ -0,0 +1,140 @@
+# Copyright The Caikit Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Standard
+from pathlib import Path
+from typing import List
+import os
+
+# Third Party
+from sentence_transformers import SentenceTransformer
+
+# First Party
+from caikit.core import ModuleBase, ModuleConfig, ModuleSaver, module
+from caikit.core.exceptions import error_handler
+import alog
+
+# Local
+from .embedding_retrieval_task import EmbeddingRetrievalTask
+from caikit_nlp.data_model.embedding_vectors import EmbeddingResult, Vector1D
+
+logger = alog.use_channel("<EMBD_BLK>")
+error = error_handler.get(logger)
+
+HOME = Path.home()
+DEFAULT_HF_MODEL = "sentence-transformers/all-MiniLM-L6-v2"
+
+
+@module(
+    "EEB12558-B4FA-4F34-A9FD-3F5890E9CD3F",
+    "EmbeddingModule",
+    "0.0.1",
+    EmbeddingRetrievalTask,
+)
+class EmbeddingModule(ModuleBase):
+    def __init__(
+        self,
+        model: SentenceTransformer,
+    ):
+        super().__init__()
+        self.model = model
+
+    @classmethod
+    def load(cls, model_path: str) -> "EmbeddingModule":
+        """Load a sequence classification model
+
+        Args:
+            model_path: str
+                Path to the config dir of the model to be loaded.
+
+        Returns:
+            EmbeddingModule
+                Instance of this class built from the model.
+        """
+
+        config = ModuleConfig.load(model_path)
+        load_path = config.get("model_artifacts")
+
+        artifact_path = False
+        if load_path:
+            if os.path.isabs(load_path) and os.path.isdir(load_path):
+                artifact_path = load_path
+            else:
+                full_path = os.path.join(model_path, load_path)
+                if os.path.isdir(full_path):
+                    artifact_path = full_path
+
+        if not artifact_path:
+            artifact_path = config.get("hf_model", DEFAULT_HF_MODEL)
```

Comment:
A little confused, how is `ht_model` value actually getting into the `config` ? This isn't getting set in `.save` method below?

Reply:
I'll have to fix save.

Reply:
So I did a bunch of fixin of save(), but re: hf_model...  How it gets in there is someone edits the config.yml which is how we usually do it because there isn't really a great mechanism that I see anywhere for bootstrap+save for non-coders [that's a different issue].   Where hf_model is particularly useful is BYO model download for HF. This is pretty common for dev and allows the cache to be used.  BUT related to this comment I did some work on save() to make it safer but also it will save the model and set artifact_path, so hf_model is kind of a bootstrap or dev mechanism.

Reply:
hf_model stuff removed